#7122 Remove examples

offtherailz opened the pull request:
## Description
- Remove examples
- Remove old application tutorial (now we have projects, this may be confusing).
- Restore api.html (https://github.com/geosolutions-it/MapStore2/issues/7079)

This PR contains changes for #7122 

Missing things are: 
 - api.html should be avaialble  in replacement of examples, for testing. 
 - Checks of build and run.
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

<!-- add here the ReadTheDocs link (if needed) -->

## Issue

**What is the current behavior?**
<!-- You can also link to an existing issue here -->
#7122
**What is the new behavior?**
<!-- Describe here the new behaviour based on your changes -->

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications -->

## Other useful information

MV88 commented on web/client/components/buttons/__tests__/ZoomButton-test.jsx:
```diff
@@ -14,7 +14,21 @@ import expect from 'expect';
 import { Provider } from 'react-redux';
 
-import store from './../../../examples/myapp/stores/myappstore';
+var {createStore, combineReducers, applyMiddleware} = require('redux');
+
+var thunkMiddleware = require('redux-thunk');
+var mapConfig = require('../../../reducers/config').default;
```
please verify these
```suggestion
import{createStore, combineReducers, applyMiddleware} from 'redux';

import thunkMiddleware from 'redux-thunk';
import mapConfig from '../../../reducers/config';
```

MV88 commented on web/client/components/buttons/__tests__/ZoomToMaxExtentButton-test.jsx:
```diff
@@ -14,7 +14,21 @@ import expect from 'expect';
 import { Provider } from 'react-redux';
 
-import store from './../../../examples/myapp/stores/myappstore';
+var {createStore, combineReducers, applyMiddleware} = require('redux');
+
+var thunkMiddleware = require('redux-thunk');
+var mapConfig = require('../../../reducers/config').default;
```
please verify these
```suggestion
import{createStore, combineReducers, applyMiddleware} from 'redux';

import thunkMiddleware from 'redux-thunk';
import mapConfig from '../../../reducers/config';
```

MV88 reviewed (changes requested):
what is the  `build/packages/framework/mapstore-framework-0.0.1.tgz` ?

in the description you mention also "restore api", but you are not doing it here so i was a bit confused

also what do you mean with "api.html should be available in replacement of examples, for testing" aren't they a 'bit' different?


Overall the PR is good

offtherailz commented on build/packages/framework/.npmignore:
```diff
@@ -0,0 +1 @@
+**/__tests__
```
Good catch. This has been added  in error,and also mapstore-framework package.

offtherailz commented on docs/developer-guide/extensions.md:
```diff
@@ -12,12 +12,11 @@ The easiest way to develop an extension is to start from the [MapStoreExtension
 
 A MapStore extension is a plugin, with some additional features.
 
-`build/extensions/plugins/SampleExtension.jsx`
```
This file doesn't exists anymore, and it has been removed a long time ago.

offtherailz commented:
> what is the `build/packages/framework/mapstore-framework-0.0.1.tgz` ?
> 
An error, removed 

> in the description you mention also "restore api", but you are not doing it here so i was a bit confused
> 
> also what do you mean with "api.html should be available in replacement of examples, for testing" aren't they a 'bit' different?
> 
There was an example, not running, that allowed to test the JS API (and so verify that is running). I removed it, but I wanted also to ensure that a tool to test api still exists. api.html now should replace this. There is a separate PR that fixes this.
https://github.com/geosolutions-it/MapStore2/pull/7133 

> Overall the PR is good

offtherailz commented on web/client/components/buttons/__tests__/ZoomToMaxExtentButton-test.jsx:
```diff
@@ -14,7 +14,21 @@ import expect from 'expect';
 import { Provider } from 'react-redux';
 
-import store from './../../../examples/myapp/stores/myappstore';
+import{createStore, combineReducers, applyMiddleware} from 'redux';
```
```suggestion
import {createStore, combineReducers, applyMiddleware} from 'redux';
```

offtherailz commented on web/client/components/buttons/__tests__/ZoomButton-test.jsx:
```diff
@@ -14,7 +14,21 @@ import expect from 'expect';
 import { Provider } from 'react-redux';
 
-import store from './../../../examples/myapp/stores/myappstore';
+import{createStore, combineReducers, applyMiddleware} from 'redux';
```
```suggestion
import {createStore, combineReducers, applyMiddleware} from 'redux';
```
